allow specifying/replacing compression level of ZipEntry

Nowheresly opened the pull request:
fix issue #61

shelajev commented:
Hi @Nowheresly, thanks for the contribution. Seems quite useful. 

Can you please provide a couple of unit-tests?

Nowheresly commented:
Hi @shelajev 
Makes sense indeed. I updated my PR accordingly !
I removed as well the crc parameter. Crc is now calculated if needed in ByteSource class.

shelajev commented:
Thanks, @Nowheresly!
